meteor/lib/api/triggers/actionFilterChainCompilers.ts

```diff
@@ -651,32 +651,54 @@ export function compileAdLibFilter(
 			}
 		}
 
-		let result: IWrappedAdLib[] = []
+		const partMap = new Map<PartId, DBPart>()
+		{
+			if (partFilter?.length) {
+				Parts.find({
```

Comment:
This will be terribly reactive and we have a lot of these filters running at the same time. For one, I think we need to use fields here to only react to the _part changing, and then, I think this needs to be wrapped into a `memoizedIsolatedAutorun` so that this is only calculated once for a given `partFilter`, which many of these filters will share.